cni.go

```diff
@@ -310,3 +311,21 @@ func (c *libcni) GetConfig() *ConfigResult {
 func (c *libcni) reset() {
 	c.networks = nil
 }
+
+func (c *libcni) Status(ctx context.Context) error {
+	err := c.Ready()
+
+	if err != nil {
+		return err
+	}
+
+	for _, network := range c.Networks() {
+		err = network.Status(ctx)
```

Comment:
this is something you may need to document, as this means if any of the networks is not ready, the status is not ready

You may want to do, if just one network is ready then status is ready 🤷

Reply:
For the cri-plugin case, we load two networks. One is the loopback plugin as separate (https://github.com/containerd/go-cni/blob/6603d5bd8941d7f2026bb5627f6aa4ff434f859a/opts.go#L88)
The other 'network' is the first CNI network configuration in /etc/cni/net.d. 

https://github.com/containerd/containerd/blob/7f0f49b43825b76e0271ed0102bf7df560826046/internal/cri/server/service_linux.go#L113

Since this is hard coded in containerd, we probably don't need to worry about this. However we could have a configurable failure behavior. If 1 fails, they all fail, if one is successful the node has the network ready condition set?

Reply:
@mikebrow thoughts?

Reply:
If one is the loopback plugin, taht is always going to be ready and not be able to provide connectivity, then it makes sense for ready to be an AND of all plugins instead of an OR, as you will need the loopback AND the other CNI to be able to provide networking to the pods

Reply:
I was wondering if this should be left up to the caller of the Status function? In this case for the containerd cri plugin I think it makes sense for AND however go-cni is used by others as well, we probably shouldn't dictate behavior?